Build error with latest git HEAD on PMIX_SERVER_SCHEDULER

## Background information
 - PRRTE master @ bfb0246
 - PMIx v3.1.3
 - Build Host:
    * Operating system/version: Linux ubuntu 16.04
    * Computer hardware:  x86-64
    * Network type: ethernet

-----------------------------

## Details of the problem

Build failure with latest PRRTE master using pmix-3.1.3 due to missing `PMIX_SERVER_SCHEDULER` in PMIx version that is unguarded in PRRTE.  Not sure if this should be a configury check or how PMIx version specific bits are handled in PRRTE.

```shell
../configure \
    --prefix=$PKG_INSTALL_PREFIX \
    --with-hwloc=$HWLOC_INSTALL_DIR \
    --with-pmix=$PMIX_INSTALL_DIR \
    --with-libevent=$LIBEVENT_INSTALL_DIR \
    --enable-orterun-prefix-by-default \
&& make -j 4 \
&& make install

...<snip>...

  CC       orted/pmix/pmix_server_pub.lo
In file included from /usr/include/string.h:630:0,
                 from /home/3t4/projects/pmix/ssd-pmix/prrte/install/include/hwloc.h:59,
                 from ../../opal/hwloc/hwloc-internal.h:28,
                 from ../../opal/util/proc.h:22,
                 from ../../orte/include/orte/types.h:30,
                 from ../../orte/orted/pmix/pmix_server.c:30:
../../orte/orted/pmix/pmix_server.c: In function ‘pmix_server_init’:
../../orte/orted/pmix/pmix_server.c:382:26: error: ‘PMIX_SERVER_SCHEDULER’ undeclared (first use in this function)
         kv->key = strdup(PMIX_SERVER_SCHEDULER);
                          ^
../../orte/orted/pmix/pmix_server.c:382:26: note: each undeclared identifier is reported only once for each function it appears in
Makefile:1473: recipe for target 'orted/pmix/pmix_server.lo' failed
```

--- a/orte/orted/pmix/pmix_server.c
+++ b/orte/orted/pmix/pmix_server.c
@@ -378,16 +378,20 @@ int pmix_server_init(void)
     /* if we are the MASTER, then we are the scheduler
      * as well as a gateway */
     if (ORTE_PROC_IS_MASTER) {
+#ifdef PMIX_SERVER_SCHEDULER
         kv = OBJ_NEW(opal_value_t);
         kv->key = strdup(PMIX_SERVER_SCHEDULER);
         kv->type = OPAL_BOOL;
         kv->data.flag = true;
         opal_list_append(&ilist, &kv->super);
+#endif
+#ifdef PMIX_SERVER_GATEWAY
         kv = OBJ_NEW(opal_value_t);
         kv->key = strdup(PMIX_SERVER_GATEWAY);
         kv->type = OPAL_BOOL;
         kv->data.flag = true;
         opal_list_append(&ilist, &kv->super);
+#endif
     }
 
     /* PRRTE always allows remote tool connections */

